port/common/omrmem32helpers.c

```diff
@@ -566,9 +566,11 @@ allocateVmemRegion32(struct OMRPortLibrary *portLibrary, uintptr_t byteAmount, J
 	}
 
 	/* use a minimum 4K page size */
-	if (pageSize < 0x1000) {
-		pageSize = 0x1000;
-	}
+	// if (pageSize < 0x1000) {
+	// 	pageSize = 0x1000;
+	// }
+	/* Make sure that we only allocate 4k pages. */
+	pageSize = 0x1000;
```

Comment:
If 4K is not one of the supported page size, we should not enforce it. While it's true that most architectures support 4K pages, we should not count on it.

Reply:
How can I know if 4K is not one of the supported page size on the architectures to avoid counting on it?

Reply:
The first entry is the default page size. The structure is setup in `omrvmem_startup`

Reply:
The code just a few lines above obtains the page of the default pages:
`pageSize = portLibrary->vmem_supported_page_sizes(portLibrary)[0];`

If this size is larger than 4K, we probably don't want to use disclaiming.

Reply:
Note, too, that this isn't enforcing anything on platforms that don't have 4K pages; it would just error out here: https://github.com/eclipse/omr/blob/a3bc75a88c9099622463fe90af1697a92e00ae41/port/linux/omrvmem.c#L861-L865

Which of our linux platform are using > 4K pages by default?

Reply:
> Which of our linux platforms are using > 4K pages by default?

From Google: "ARM processors can have 64k pages, and some distributions, like CentOS, RedHat, or Oracle Linux, set the default page size to 64k"

Reply:
It seems like if 4K is not available then I will look for the smallest one as per `omrvmem_startup`, even though if this size is larger than 4K we probably don't want to use disclaiming on machines that set the default page size to 64k.